GitUI/BranchTreePanel/RepoObjectsTree.ContextActions.cs

```diff
@@ -151,5 +167,26 @@ private void contextMenu_Opening(object sender, CancelEventArgs e)
             ContextMenuBranchSpecific(contextMenu);
             ContextMenuRemoteRepoSpecific(contextMenu);
         }
+
+        /// <inheritdoc />
+        public TMenuItem CreateMenuItem<TMenuItem, TNode>(Action<TNode> onClick, string captionKey, string toolTipKey, Bitmap icon = null)
+            where TMenuItem : ToolStripItem, new()
+            where TNode : class, INode
+        {
+            var result = new TMenuItem();
+            result.Image = icon;
+            result.Text = new TranslationString(captionKey).Text;
+            result.ToolTipText = new TranslationString(toolTipKey).Text;
+            RegisterClick(result, onClick);
+            return result;
+        }
+
+        private void AddContextMenuItems(IEnumerable<ToolStripItem> items, ContextMenuStrip menu, ToolStripItem insertAfter = null)
+        {
+            menu.SuspendLayout();
+            int index = insertAfter == null ? 0 : Math.Max(0, menu.Items.IndexOf(insertAfter) + 1);
+            items.Reverse().ForEach(item => menu.Items.Insert(index, item));
```

Comment:
It feels like a misplaced responsibility to reverse the menu items here.
Why aren't items supplied in the desired order? Shouldn't the underlying collections be sorted in the first place?

Overall this method feels like it could be turned into an extension method and made generally available for other context menus.

Reply:
The items are supplied in the right order. It's the insert at the same index that needs reverse. I'll changed this to increment index to avoid Reverse 😄